packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts

```diff
@@ -0,0 +1,935 @@
+    const domainName = 'test-domain-2w2x2u3tifly';
+    const domainEndpoint = `https://${domainName}-jcjotrt6f7otem4sqcwbch3c4u.testregion.es.amazonaws.com`;
+    const imported = Domain.fromDomainEndpoint(stack, 'Domain', domainEndpoint);
+
+    expect(imported.domainName).toEqual(domainName);
+    expect(imported.domainArn).toMatch(RegExp(`es:testregion:1234:domain/${domainName}$`));
+
+    expect(stack).not.toHaveResource('AWS::Elasticsearch::Domain');
+  });
+
+  test('static fromDomainAttributes(attributes) allows importing an external/existing domain', () => {
+    const domainName = 'test-domain-2w2x2u3tifly';
+    const domainArn = `es:testregion:1234:domain/${domainName}`;
+    const domainEndpoint = `https://${domainName}-jcjotrt6f7otem4sqcwbch3c4u.testregion.es.amazonaws.com`;
+    const imported = Domain.fromDomainAttributes(stack, 'Domain', {
+      domainArn,
+      domainEndpoint,
+    });
+
+    expect(imported.domainName).toEqual(domainName);
+    expect(imported.domainArn).toEqual(domainArn);
+
+    expect(stack).not.toHaveResource('AWS::Elasticsearch::Domain');
+  });
+
+});
+
+describe('advanced security options', () => {
+  const masterUserArn = 'arn:aws:iam::123456789012:user/JohnDoe';
+  const masterUserName = 'JohnDoe';
+  const password = 'password';
+  const masterUserPassword = SecretValue.plainText(password);
+
+  test('enable fine-grained access logging with a master user ARN', () => {
+    new Domain(stack, 'Domain', {
+      version: ElasticsearchVersion.V7_1,
+      fineGrainedAccessControl: {
+        masterUserArn,
+      },
+      encryptionAtRest: {
+        enabled: true,
+      },
+      nodeToNodeEncryption: true,
+      enforceHttps: true,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::Elasticsearch::Domain', {
+      AdvancedSecurityOptions: {
+        Enabled: true,
+        InternalUserDatabaseEnabled: false,
+        MasterUserOptions: {
+          MasterUserARN: masterUserArn,
+        },
+      },
+      EncryptionAtRestOptions: {
+        Enabled: true,
+      },
+      NodeToNodeEncryptionOptions: {
+        Enabled: true,
+      },
+      DomainEndpointOptions: {
+        EnforceHTTPS: true,
+      },
+    });
+  });
+
+  test('enable fine-grained access logging with a master user name and password', () => {
+    new Domain(stack, 'Domain', {
+      version: ElasticsearchVersion.V7_1,
+      fineGrainedAccessControl: {
+        masterUserName,
+        masterUserPassword,
+      },
+      encryptionAtRest: {
+        enabled: true,
+      },
+      nodeToNodeEncryption: true,
+      enforceHttps: true,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::Elasticsearch::Domain', {
+      AdvancedSecurityOptions: {
+        Enabled: true,
+        InternalUserDatabaseEnabled: true,
+        MasterUserOptions: {
+          MasterUserName: masterUserName,
+          MasterUserPassword: password,
```

Comment:
Is this working as expected or should the plain text password be more obfuscated?

Reply:
Yeap thats fine